mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -184,7 +241,23 @@ def transformation():  # pylint: disable=unused-variable
             data = flask.request.data.decode("utf-8")
             csv_input = StringIO(data)
             data = parse_csv_input(csv_input=csv_input)
-        elif flask.request.content_type in [CONTENT_TYPE_JSON, CONTENT_TYPE_JSON_SPLIT_ORIENTED]:
+        elif flask.request.content_type == CONTENT_TYPE_JSON:
+            json_str = flask.request.data.decode("utf-8")
+            try:
+                inp_dict = json.loads(json_str)
+            except json.decoder.JSONDecodeError:
+                _handle_serving_error(
+                    error_message=(
+                        "Failed to parse input from JSON. Ensure that input is a valid JSON"
+                        " formatted string."
+                    ),
+                    error_code=MALFORMED_REQUEST,
+                )
+            if "instances" in inp_dict or "inputs" in inp_dict:
+                data = parse_tf_serving_input(inp_dict)
+            else:
+                data = parse_json_input(json_input=json_str, orient="split", schema=input_schema)
```

Comment:
we also want to infer the correct orient value in this case based on the input type, list -> orient='records', dict with at least 'values' attribute -> orient='split'.